# Question #3 What's the probability that a patient who has COVID will be recovered vs passing away(death)?


### Importing Libraries

In [1]:
import pandas as pd
import numpy as np

### Reading Data file

In [2]:
# Death Sata Set

dfA_0=pd.read_csv("time_series_covid19_deaths_global_03-31-2020.csv")
dfA_0

Province/State  Country/Region        Lat       Long  1/22/20  \
0                         NaN     Afghanistan  33.000000  65.000000        0   
1                         NaN         Albania  41.153300  20.168300        0   
2                         NaN         Algeria  28.033900   1.659600        0   
3                         NaN         Andorra  42.506300   1.521800        0   
4                         NaN          Angola -11.202700  17.873900        0   
..                        ...             ...        ...        ...      ...   
251  Turks and Caicos Islands  United Kingdom  21.694000 -71.797900        0   
252                       NaN      MS Zaandam   0.000000   0.000000        0   
253                       NaN        Botswana -22.328500  24.684900        0   
254                       NaN         Burundi  -3.373100  29.918900        0   
255                       NaN    Sierra Leone   8.460555 -11.779889        0   

     1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  ...  3/22/20  3/23/20  \
0          0        0        0        0        0  ...        1        1   
1          0        0        0        0        0  ...        2        4   
2          0        0        0        0        0  ...       17       17   
3          0        0        0        0        0  ...        1        1   
4          0        0        0        0        0  ...        0        0   
..       ...      ...      ...      ...      ...  ...      ...      ...   
251        0        0        0        0        0  ...        0        0   
252        0        0        0        0        0  ...        0        0   
253        0        0        0        0        0  ...        0        0   
254        0        0        0        0        0  ...        0        0   
255        0        0        0        0        0  ...        0        0   

     3/24/20  3/25/20  3/26/20  3/27/20  3/28/20  3/29/20  3/30/20  3/31/20  
0          1        2        4        4        4        4        4        4  
1          5        5        6        8       10       10       11       15  
2         19       21       25       26       29       31       35       44  
3          1        1        3        3        3        6        8       12  
4          0        0        0        0        0        2        2        2  
..       ...      ...      ...      ...      ...      ...      ...      ...  
251        0        0        0        0        0        0        0        0  
252        0        0        0        0        0        0        0        0  
253        0        0        0        0        0        0        0        1  
254        0        0        0        0        0        0        0        0  
255        0        0        0        0        0        0        0        0  

[256 rows x 74 columns]

In [7]:
# Recovery Datasets

dfB_0=pd.read_csv("time_series_covid19_recovered_global_03-31-2020.csv")
dfB_0

Province/State  Country/Region        Lat       Long  1/22/20  \
0                         NaN     Afghanistan  33.000000  65.000000        0   
1                         NaN         Albania  41.153300  20.168300        0   
2                         NaN         Algeria  28.033900   1.659600        0   
3                         NaN         Andorra  42.506300   1.521800        0   
4                         NaN          Angola -11.202700  17.873900        0   
..                        ...             ...        ...        ...      ...   
237  Turks and Caicos Islands  United Kingdom  21.694000 -71.797900        0   
238                       NaN      MS Zaandam   0.000000   0.000000        0   
239                       NaN        Botswana -22.328500  24.684900        0   
240                       NaN         Burundi  -3.373100  29.918900        0   
241                       NaN    Sierra Leone   8.460555 -11.779889        0   

     1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  ...  3/22/20  3/23/20  \
0          0        0        0        0        0  ...        1        1   
1          0        0        0        0        0  ...        2        2   
2          0        0        0        0        0  ...       65       65   
3          0        0        0        0        0  ...        1        1   
4          0        0        0        0        0  ...        0        0   
..       ...      ...      ...      ...      ...  ...      ...      ...   
237        0        0        0        0        0  ...        0        0   
238        0        0        0        0        0  ...        0        0   
239        0        0        0        0        0  ...        0        0   
240        0        0        0        0        0  ...        0        0   
241        0        0        0        0        0  ...        0        0   

     3/24/20  3/25/20  3/26/20  3/27/20  3/28/20  3/29/20  3/30/20  3/31/20  
0          1        2        2        2        2        2        2        5  
1         10       17       17       31       31       33       44       52  
2         24       65       29       29       31       31       37       46  
3          1        1        1        1        1        1       10       10  
4          0        0        0        0        0        0        0        1  
..       ...      ...      ...      ...      ...      ...      ...      ...  
237        0        0        0        0        0        0        0        0  
238        0        0        0        0        0        0        0        0  
239        0        0        0        0        0        0        0        0  
240        0        0        0        0        0        0        0        0  
241        0        0        0        0        0        0        0        0  

[242 rows x 74 columns]

In [11]:
# Confirmed Case Datasets

dfC_0=pd.read_csv("time_series_covid19_confirmed_global_03-31-20.csv")
dfC_0

Province/State  Country/Region        Lat       Long  1/22/20  \
0                         NaN     Afghanistan  33.000000  65.000000        0   
1                         NaN         Albania  41.153300  20.168300        0   
2                         NaN         Algeria  28.033900   1.659600        0   
3                         NaN         Andorra  42.506300   1.521800        0   
4                         NaN          Angola -11.202700  17.873900        0   
..                        ...             ...        ...        ...      ...   
251  Turks and Caicos Islands  United Kingdom  21.694000 -71.797900        0   
252                       NaN      MS Zaandam   0.000000   0.000000        0   
253                       NaN        Botswana -22.328500  24.684900        0   
254                       NaN         Burundi  -3.373100  29.918900        0   
255                       NaN    Sierra Leone   8.460555 -11.779889        0   

     1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  ...  3/22/20  3/23/20  \
0          0        0        0        0        0  ...       40       40   
1          0        0        0        0        0  ...       89      104   
2          0        0        0        0        0  ...      201      230   
3          0        0        0        0        0  ...      113      133   
4          0        0        0        0        0  ...        2        3   
..       ...      ...      ...      ...      ...  ...      ...      ...   
251        0        0        0        0        0  ...        0        0   
252        0        0        0        0        0  ...        0        0   
253        0        0        0        0        0  ...        0        0   
254        0        0        0        0        0  ...        0        0   
255        0        0        0        0        0  ...        0        0   

     3/24/20  3/25/20  3/26/20  3/27/20  3/28/20  3/29/20  3/30/20  3/31/20  
0         74       84       94      110      110      120      170      174  
1        123      146      174      186      197      212      223      243  
2        264      302      367      409      454      511      584      716  
3        164      188      224      267      308      334      370      376  
4          3        3        4        4        5        7        7        7  
..       ...      ...      ...      ...      ...      ...      ...      ...  
251        0        0        0        0        4        4        5        5  
252        0        0        0        0        2        2        2        2  
253        0        0        0        0        0        0        3        4  
254        0        0        0        0        0        0        0        2  
255        0        0        0        0        0        0        0        1  

[256 rows x 74 columns]

In [12]:
dfC_0.columns

Index(['Province/State', 'Country/Region', 'Lat', 'Long', '1/22/20', '1/23/20',
       '1/24/20', '1/25/20', '1/26/20', '1/27/20', '1/28/20', '1/29/20',
       '1/30/20', '1/31/20', '2/1/20', '2/2/20', '2/3/20', '2/4/20', '2/5/20',
       '2/6/20', '2/7/20', '2/8/20', '2/9/20', '2/10/20', '2/11/20', '2/12/20',
       '2/13/20', '2/14/20', '2/15/20', '2/16/20', '2/17/20', '2/18/20',
       '2/19/20', '2/20/20', '2/21/20', '2/22/20', '2/23/20', '2/24/20',
       '2/25/20', '2/26/20', '2/27/20', '2/28/20', '2/29/20', '3/1/20',
       '3/2/20', '3/3/20', '3/4/20', '3/5/20', '3/6/20', '3/7/20', '3/8/20',
       '3/9/20', '3/10/20', '3/11/20', '3/12/20', '3/13/20', '3/14/20',
       '3/15/20', '3/16/20', '3/17/20', '3/18/20', '3/19/20', '3/20/20',
       '3/21/20', '3/22/20', '3/23/20', '3/24/20', '3/25/20', '3/26/20',
       '3/27/20', '3/28/20', '3/29/20', '3/30/20', '3/31/20'],
      dtype='object')

### Making a Dataframe of our required Dates and summing

In [6]:
# Death Data set
dfA_1=dfA_0[['Province/State', 'Country/Region', 'Lat', 'Long','2/1/20', '2/2/20', '2/3/20', '2/4/20', '2/5/20',
       '2/6/20', '2/7/20', '2/8/20', '2/9/20', '2/10/20', '2/11/20', '2/12/20',
       '2/13/20', '2/14/20', '2/15/20', '2/16/20', '2/17/20', '2/18/20',
       '2/19/20', '2/20/20', '2/21/20', '2/22/20', '2/23/20', '2/24/20',
       '2/25/20', '2/26/20', '2/27/20', '2/28/20', '2/29/20', '3/1/20',
       '3/2/20', '3/3/20', '3/4/20', '3/5/20', '3/6/20', '3/7/20', '3/8/20',
       '3/9/20', '3/10/20', '3/11/20', '3/12/20', '3/13/20', '3/14/20',
       '3/15/20', '3/16/20', '3/17/20', '3/18/20', '3/19/20', '3/20/20',
       '3/21/20', '3/22/20', '3/23/20', '3/24/20', '3/25/20', '3/26/20',
       '3/27/20', '3/28/20', '3/29/20', '3/30/20', '3/31/20']]

dfA_2=dfA_1[:]
dfA_2['Sum_Death_feb_March']=dfA_1.iloc[:,5:].sum(axis=1)
dfA_2

C:\ProgramData\Anaconda3\lib\site-packages\ipykernel_launcher.py:14: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


Province/State  Country/Region        Lat       Long  2/1/20  \
0                         NaN     Afghanistan  33.000000  65.000000       0   
1                         NaN         Albania  41.153300  20.168300       0   
2                         NaN         Algeria  28.033900   1.659600       0   
3                         NaN         Andorra  42.506300   1.521800       0   
4                         NaN          Angola -11.202700  17.873900       0   
..                        ...             ...        ...        ...     ...   
251  Turks and Caicos Islands  United Kingdom  21.694000 -71.797900       0   
252                       NaN      MS Zaandam   0.000000   0.000000       0   
253                       NaN        Botswana -22.328500  24.684900       0   
254                       NaN         Burundi  -3.373100  29.918900       0   
255                       NaN    Sierra Leone   8.460555 -11.779889       0   

     2/2/20  2/3/20  2/4/20  2/5/20  2/6/20  ...  3/23/20  3/24/20  3/25/20  \
0         0       0       0       0       0  ...        1        1        2   
1         0       0       0       0       0  ...        4        5        5   
2         0       0       0       0       0  ...       17       19       21   
3         0       0       0       0       0  ...        1        1        1   
4         0       0       0       0       0  ...        0        0        0   
..      ...     ...     ...     ...     ...  ...      ...      ...      ...   
251       0       0       0       0       0  ...        0        0        0   
252       0       0       0       0       0  ...        0        0        0   
253       0       0       0       0       0  ...        0        0        0   
254       0       0       0       0       0  ...        0        0        0   
255       0       0       0       0       0  ...        0        0        0   

     3/26/20  3/27/20  3/28/20  3/29/20  3/30/20  3/31/20  Sum_Death_feb_March  
0          4        4        4        4        4        4                   29  
1          6        8       10       10       11       15                   91  
2         25       26       29       31       35       44                  324  
3          3        3        3        6        8       12                   39  
4          0        0        0        2        2        2                    6  
..       ...      ...      ...      ...      ...      ...                  ...  
251        0        0        0        0        0        0                    0  
252        0        0        0        0        0        0                    0  
253        0        0        0        0        0        1                    1  
254        0        0        0        0        0        0                    0  
255        0        0        0        0        0        0                    0  

[256 rows x 65 columns]

In [8]:
# Recovery Datasets

dfB_1=dfB_0[['Province/State', 'Country/Region', 'Lat', 'Long', 
             '2/1/20', '2/2/20', '2/3/20', '2/4/20', '2/5/20',
       '2/6/20', '2/7/20', '2/8/20', '2/9/20', '2/10/20', '2/11/20', '2/12/20',
       '2/13/20', '2/14/20', '2/15/20', '2/16/20', '2/17/20', '2/18/20',
       '2/19/20', '2/20/20', '2/21/20', '2/22/20', '2/23/20', '2/24/20',
       '2/25/20', '2/26/20', '2/27/20', '2/28/20', '2/29/20', '3/1/20',
       '3/2/20', '3/3/20', '3/4/20', '3/5/20', '3/6/20', '3/7/20', '3/8/20',
       '3/9/20', '3/10/20', '3/11/20', '3/12/20', '3/13/20', '3/14/20',
       '3/15/20', '3/16/20', '3/17/20', '3/18/20', '3/19/20', '3/20/20',
       '3/21/20', '3/22/20', '3/23/20', '3/24/20', '3/25/20', '3/26/20',
       '3/27/20', '3/28/20', '3/29/20', '3/30/20', '3/31/20']]

dfB_2=dfB_1[:]
dfB_2['Sum_Recovered_feb_March']=dfB_1.iloc[:,5:].sum(axis=1)
dfB_2

C:\ProgramData\Anaconda3\lib\site-packages\ipykernel_launcher.py:16: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  app.launch_new_instance()


Province/State  Country/Region        Lat       Long  2/1/20  \
0                         NaN     Afghanistan  33.000000  65.000000       0   
1                         NaN         Albania  41.153300  20.168300       0   
2                         NaN         Algeria  28.033900   1.659600       0   
3                         NaN         Andorra  42.506300   1.521800       0   
4                         NaN          Angola -11.202700  17.873900       0   
..                        ...             ...        ...        ...     ...   
237  Turks and Caicos Islands  United Kingdom  21.694000 -71.797900       0   
238                       NaN      MS Zaandam   0.000000   0.000000       0   
239                       NaN        Botswana -22.328500  24.684900       0   
240                       NaN         Burundi  -3.373100  29.918900       0   
241                       NaN    Sierra Leone   8.460555 -11.779889       0   

     2/2/20  2/3/20  2/4/20  2/5/20  2/6/20  ...  3/23/20  3/24/20  3/25/20  \
0         0       0       0       0       0  ...        1        1        2   
1         0       0       0       0       0  ...        2       10       17   
2         0       0       0       0       0  ...       65       24       65   
3         0       0       0       0       0  ...        1        1        1   
4         0       0       0       0       0  ...        0        0        0   
..      ...     ...     ...     ...     ...  ...      ...      ...      ...   
237       0       0       0       0       0  ...        0        0        0   
238       0       0       0       0       0  ...        0        0        0   
239       0       0       0       0       0  ...        0        0        0   
240       0       0       0       0       0  ...        0        0        0   
241       0       0       0       0       0  ...        0        0        0   

     3/26/20  3/27/20  3/28/20  3/29/20  3/30/20  3/31/20  \
0          2        2        2        2        2        5   
1         17       31       31       33       44       52   
2         29       29       31       31       37       46   
3          1        1        1        1       10       10   
4          0        0        0        0        0        1   
..       ...      ...      ...      ...      ...      ...   
237        0        0        0        0        0        0   
238        0        0        0        0        0        0   
239        0        0        0        0        0        0   
240        0        0        0        0        0        0   
241        0        0        0        0        0        0   

     Sum_Recovered_feb_March  
0                         26  
1                        241  
2                        594  
3                         37  
4                          1  
..                       ...  
237                        0  
238                        0  
239                        0  
240                        0  
241                        0  

[242 rows x 65 columns]

In [14]:
# Confirmed Cases Datasets

dfC_1=dfC_0[['Province/State', 'Country/Region', 'Lat', 'Long', 
             '2/1/20', '2/2/20', '2/3/20', '2/4/20', '2/5/20',
       '2/6/20', '2/7/20', '2/8/20', '2/9/20', '2/10/20', '2/11/20', '2/12/20',
       '2/13/20', '2/14/20', '2/15/20', '2/16/20', '2/17/20', '2/18/20',
       '2/19/20', '2/20/20', '2/21/20', '2/22/20', '2/23/20', '2/24/20',
       '2/25/20', '2/26/20', '2/27/20', '2/28/20', '2/29/20', '3/1/20',
       '3/2/20', '3/3/20', '3/4/20', '3/5/20', '3/6/20', '3/7/20', '3/8/20',
       '3/9/20', '3/10/20', '3/11/20', '3/12/20', '3/13/20', '3/14/20',
       '3/15/20', '3/16/20', '3/17/20', '3/18/20', '3/19/20', '3/20/20',
       '3/21/20', '3/22/20', '3/23/20', '3/24/20', '3/25/20', '3/26/20',
       '3/27/20', '3/28/20', '3/29/20', '3/30/20', '3/31/20']]

dfC_2=dfC_1[:]
dfC_2['Sum_Confirmed_feb_March']=dfC_1.iloc[:,5:].sum(axis=1)
dfC_2

C:\ProgramData\Anaconda3\lib\site-packages\ipykernel_launcher.py:16: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  app.launch_new_instance()


Province/State  Country/Region        Lat       Long  2/1/20  \
0                         NaN     Afghanistan  33.000000  65.000000       0   
1                         NaN         Albania  41.153300  20.168300       0   
2                         NaN         Algeria  28.033900   1.659600       0   
3                         NaN         Andorra  42.506300   1.521800       0   
4                         NaN          Angola -11.202700  17.873900       0   
..                        ...             ...        ...        ...     ...   
251  Turks and Caicos Islands  United Kingdom  21.694000 -71.797900       0   
252                       NaN      MS Zaandam   0.000000   0.000000       0   
253                       NaN        Botswana -22.328500  24.684900       0   
254                       NaN         Burundi  -3.373100  29.918900       0   
255                       NaN    Sierra Leone   8.460555 -11.779889       0   

     2/2/20  2/3/20  2/4/20  2/5/20  2/6/20  ...  3/23/20  3/24/20  3/25/20  \
0         0       0       0       0       0  ...       40       74       84   
1         0       0       0       0       0  ...      104      123      146   
2         0       0       0       0       0  ...      230      264      302   
3         0       0       0       0       0  ...      133      164      188   
4         0       0       0       0       0  ...        3        3        3   
..      ...     ...     ...     ...     ...  ...      ...      ...      ...   
251       0       0       0       0       0  ...        0        0        0   
252       0       0       0       0       0  ...        0        0        0   
253       0       0       0       0       0  ...        0        0        0   
254       0       0       0       0       0  ...        0        0        0   
255       0       0       0       0       0  ...        0        0        0   

     3/26/20  3/27/20  3/28/20  3/29/20  3/30/20  3/31/20  \
0         94      110      110      120      170      174   
1        174      186      197      212      223      243   
2        367      409      454      511      584      716   
3        224      267      308      334      370      376   
4          4        4        5        7        7        7   
..       ...      ...      ...      ...      ...      ...   
251        0        0        4        4        5        5   
252        0        0        2        2        2        2   
253        0        0        0        0        3        4   
254        0        0        0        0        0        2   
255        0        0        0        0        0        1   

     Sum_Confirmed_feb_March  
0                       1225  
1                       2232  
2                       4828  
3                       2787  
4                         48  
..                       ...  
251                       18  
252                        8  
253                        7  
254                        2  
255                        1  

[256 rows x 65 columns]

### Total Confirmed Cases Datasets

In [26]:
Total_COVID_19_confirmed=dfC_2['Sum_Confirmed_feb_March'].sum(axis=0)
Total_COVID_19_confirmed

10553803

### Total Recovered Cases Datasets

In [25]:
Total_COVID_19_Recoverd=dfB_2['Sum_Recovered_feb_March'].sum(axis=0)
Total_COVID_19_Recoverd

3086599

### Total Death Cases Datasets

In [23]:
Total_COVID_19_Death=dfA_2['Sum_Death_feb_March'].sum(axis=0)
Total_COVID_19_Death

432849

## Probability that a patient who has COVID will be recovered - P(recover)

In [29]:
P_recover=Total_COVID_19_Recoverd/(Total_COVID_19_confirmed)
P_recover

0.2924632002321817

## Probability that a patient who has COVID will Die - P(death)

In [30]:
P_death=Total_COVID_19_Death/(Total_COVID_19_confirmed)
P_death

0.04101355691403374

# ANSWER 3: Probability that a patient who has COVID will be recovered vs passing away(death)?

##### Probability of Recovery is 0.29 which is higher than Probability of Dying from CoronaVirus 0.041